pandas/core/series.py

```diff
@@ -2030,7 +2033,33 @@ def autocorr(self, lag=1):
 
         Returns
         -------
-        autocorr : float
+        float
+            The Pearson correlation between self and self.shift(lag).
+
+        See also
+        --------
+        Series.corr : Compute the correlation between two Series.
+
+        Notes
+        -----
+        return Nan if the computation of the autocorr is not possible.
+
+        Examples
+        --------
+        >>> s = pd.Series([0.25, 0.5, 0.2, -0.05])
+        >>> s.autocorr()
+        0.1035526330902407
+        >>> s.autocorr(lag=2)
+        -0.9999999999999999
+
+        **Warning**
+
+        If the pearson correlation between self and self.shift(lag) cannot be
```

Comment:
Rather than "self" and "self.shift(lag)", how about "If the Pearson correlation between the two points is not defined, then \`\`NaN\`\` is returned.